internal/consensus/state.go

```diff
@@ -1713,8 +1713,9 @@ func (cs *State) recordMetrics(height int64, block *types.Block) {
 			address    types.Address
 		)
 		if commitSize != valSetLen {
-			panic(fmt.Sprintf("commit size (%d) doesn't match valset length (%d) at height %d\n\n%v\n\n%v",
+			cs.Logger.Error(fmt.Sprintf("commit size (%d) doesn't match valset length (%d) at height %d\n\n%v\n\n%v",
```

Comment:
why did we change this?

Reply:
base on the discussion here 
https://github.com/tendermint/tendermint/pull/6590#discussion_r652990078

Or we move the sanity check before call  `recordMetrics`?

BTW, I need to add a `return` here.

Reply:
Ahh I didn't realize this is a metrics function. This is OK 👍